poly-commit/src/kzg10/data_structures.rs

```diff
@@ -324,12 +325,19 @@ impl<E: Pairing> PreparedVerifierKey<E> {
     PartialEq(bound = ""),
     Eq(bound = "")
 )]
-pub struct Commitment<E: Pairing>(
+pub struct Commitment<E>(
     /// The commitment is a group element.
     pub E::G1Affine,
-);
+)
+where
+    E: Pairing,
+    E::G1Affine: Absorb;
```

Comment:
Technically, we don't need an `Absorb` bound on `E::G1Affine`, only on the `PCCommitment` impl and on the `Absorb` impl, right?

Reply:
That would avoid most of this churn I think

Reply:
If I restrict `PCCommitment: Absorb`, then I also have to ensure any struct which implements `PCCommitment` also implements `Absorb` - and so I need to restrict `E::G1Affine` etc. to also be `Absorb`.
Let me know if you have another idea for this, I currently can't see a better workaround.